js/style/style_transition.js

```diff
@@ -40,45 +38,59 @@ class StyleTransition {
     }
 
     /*
-     * Return the value of the transitioning property at zoom level `z` and optional time `t`
+     * Return the value of the transitioning property.
      */
-    calculate(globalProperties, featureProperties) {
-        let value = this.declaration.calculate(
-            util.extend({}, globalProperties, {duration: this.duration}),
-            featureProperties
-        );
-
-        if (this.instant()) return value;
-
-        const t = globalProperties.time || Date.now();
-
-        if (t < this.endTime) {
-            const oldValue = this.oldTransition.calculate(
-                util.extend({}, globalProperties, {time: this.startTime}),
-                featureProperties
-            );
-            const eased = this.ease((t - this.startTime - this.delay) / this.duration);
-            value = this.interp(oldValue, value, eased);
-        }
+    calculate(globalProperties, featureProperties, time) {
+        const value = this._calculateTargetValue(globalProperties, featureProperties);
+
+        if (this.instant())
+            return value;
+
+        time = time || Date.now();
+
+        if (time >= this.endTime)
+            return value;
 
-        return value;
+        const oldValue = this.oldTransition.calculate(globalProperties, featureProperties, this.startTime);
+        const t = util.easeCubicInOut((time - this.startTime - this.delay) / this.duration);
+        return this.interp(oldValue, value, t);
+    }
+
+    _calculateTargetValue(globalProperties, featureProperties) {
```

Comment:
This code earlier lived mostly in `style_declaration` `wrapTransitionedCalculate` [here](https://github.com/mapbox/mapbox-gl-js/blob/master/js/style/style_declaration.js#L58-L96). What this method does is calculate the value the property is _transitioning to_, as opposed to `calculate` which returns the current value that is transitioning from the old value to the target one with time — please see `calculate` above.